xarray/tests/test_backends_datatree.py

```diff
@@ -502,6 +528,26 @@ def test_open_groups(self, unaligned_datatree_zarr) -> None:
         for ds in unaligned_dict_of_datasets.values():
             ds.close()
 
+    def test_open_datatree_specific_group(self, tmpdir, simple_datatree) -> None:
+        """Test opening a specific group within a Zarr store using `open_datatree`."""
+        filepath = tmpdir / "test.zarr"
+        group = "/set2"
+        original_dt = simple_datatree
+        original_dt.to_zarr(filepath)
+
+        with open_datatree(filepath, group=group, engine=self.engine) as subgroup_tree:
+            assert subgroup_tree is not None
+            assert subgroup_tree.root.parent is None
+            assert list(subgroup_tree.children) == list(original_dt[group].children)
+            assert len(subgroup_tree.dataset.dims) == len(
+                original_dt[group].dataset.dims
+            )
+            assert list(subgroup_tree.dataset.data_vars) == list(
+                original_dt[group].dataset.data_vars
+            )
+            assert_equal(subgroup_tree.dataset["a"], original_dt[group].dataset["a"])
+            assert_equal(subgroup_tree.dataset["b"], original_dt[group].dataset["b"])
```

Comment:
is there a reason why we can't simply compare the entire dataset?
```suggestion
            assert_equal(subgroup_tree.dataset, original_dt[group].dataset)
```
(if not, `assert_equal` will give use nicer error messages)